internal/vppinit/apiparams/apiparams.go

```diff
@@ -0,0 +1,153 @@
+	var strs []string
+	strs = append(strs, fmt.Sprintf("%+v", c.AfPacket))
+	_, _ = sb.WriteString(strings.Join(strs, " "))
+	_, _ = sb.WriteString("},")
+
+	_, _ = sb.WriteString("AF_XDP:{")
+	var xdpStrs []string
+	xdpStrs = append(xdpStrs, fmt.Sprintf("%+v", c.AfXdp))
+	_, _ = sb.WriteString(strings.Join(xdpStrs, " "))
+	_, _ = sb.WriteString("},")
+	_, _ = sb.WriteString("}")
+	return sb.String()
+}
+
+// DumpToFile Dump the structure to a given file in yaml format
+func (c *ConfigYAML) DumpToFile(filename string) error {
+	contents, err := yaml.Marshal(c)
+	if err != nil {
+		return err
+	}
+	if err := os.MkdirAll(path.Dir(filename), 0o700); err != nil {
+		return err
+	}
+	return os.WriteFile(filename, append([]byte(fileStart), contents...), 0o600)
+}
+
+// AfPacketParams contains configuration parameters for AF_PACKET interface
+type AfPacketParams struct {
+	Mode             af_packet.AfPacketMode  `yaml:"mode"`
+	RxFrameSize      uint32                  `yaml:"rxFrameSize"`
+	TxFrameSize      uint32                  `yaml:"txFrameSize"`
+	RxFramesPerBlock uint32                  `yaml:"rxFramesPerBlock"`
+	TxFramesPerBlock uint32                  `yaml:"txFramesPerBlock"`
+	NumRxQueues      uint16                  `yaml:"numRxQueues"`
+	NumTxQueues      uint16                  `yaml:"numTxQueues"`
+	Flags            af_packet.AfPacketFlags `yaml:"flags"`
+}
+
+// AfXDPParams contains configuration parameters for AF_XDP interface
+type AfXDPParams struct {
+	Mode    af_xdp.AfXdpMode `yaml:"mode"`
+	RxqSize uint16           `yaml:"rxqSize"`
+	TxqSize uint16           `yaml:"txqSize"`
+	Flags   af_xdp.AfXdpFlag `yaml:"flags"`
+}
+
+// GetDefaults  Get default arguments used by create host interface APIs
+func GetDefaults() *ConfigYAML {
+	return &ConfigYAML{
+		AfPacket: &AfPacketParams{
+			Mode:             af_packet.AF_PACKET_API_MODE_ETHERNET,
+			RxFrameSize:      10240,
+			TxFrameSize:      10240,
+			RxFramesPerBlock: 1024,
+			TxFramesPerBlock: 1024,
+			NumRxQueues:      1,
+			NumTxQueues:      0,
+			Flags:            af_packet.AF_PACKET_API_FLAG_VERSION_2,
+		},
+		AfXdp: &AfXDPParams{
+			Mode:    af_xdp.AF_XDP_API_MODE_AUTO,
+			RxqSize: 8192,
+			TxqSize: 8192,
+			Flags:   0,
+		},
+	}
+}
+
+// GetAfPacketValues get parameter values for af-packet interface creation
+func GetAfPacketValues(ctx context.Context) *AfPacketParams {
+	return getConfig(ctx).AfPacket
+}
+
+// GetAfXdpValues get parameter values for af-xdp interface creation
+func GetAfXdpValues(ctx context.Context) *AfXDPParams {
+	return getConfig(ctx).AfXdp
+}
+
+func getConfig(ctx context.Context) *ConfigYAML {
+	cfg := GetDefaults()
+	if _, err := os.Stat(confFilename); os.IsNotExist(err) {
+		log.FromContext(ctx).Infof("Configuration file: %q not found, using defaults(%+v)", confFilename, cfg)
+		if err = cfg.DumpToFile(confFilename); err != nil {
+			log.FromContext(ctx).Warnf("Failed to expose used vppapi AF_ interface default values %+v", err)
+		}
```

Comment:
Could we remove DumpToFile? Seems like we could simply return cfg in the case when file doesn't exist.

```suggestion
		log.FromContext(ctx).Infof("Configuration file: %q not found, using defaults(%+v)", confFilename, cfg)
```

Reply:
`DumpToFile()` function removed and the test refactored.